IronPython3Extension/Extension.cs

```diff
@@ -29,9 +35,78 @@ public void Dispose()
             
         }
 
-        public void Ready(ReadyParams sp)
+        /// <summary>
+        /// Action to be invoked when the Dynamo has started up and is ready
+        /// for user interaction. 
+        /// </summary>
+        /// <param name="rp"></param>
+        public void Ready(ReadyParams rp)
+        {
+            var extraPath = Path.Combine(new FileInfo(Assembly.GetExecutingAssembly().Location).Directory.Parent.FullName, "extra");
+            var alc = new IsolatedPythonContext(Path.Combine(extraPath, $"{PythonEvaluatorAssembly}.dll"));
+            var dsIronAssem = alc.LoadFromAssemblyName(new AssemblyName(PythonEvaluatorAssembly));
+
+            //load the engine into Dynamo ourselves.
+            LoadPythonEngine(dsIronAssem);
+
+            //we used to do this:
+            //but it's not neccesary to load anything into the VM.
+            //instead we skip all the extra work and trigger the side effect we want
+            //which is re executing the graph after the dsIronPython evaluator is loaded into the PythonEngineManager.
+            //rp.StartupParams.LibraryLoader.LoadNodeLibrary(dsIronAssem);
+
+            if (rp.CurrentWorkspaceModel is HomeWorkspaceModel hwm)
+            {
+                foreach (var n in hwm.Nodes)
+                {
+                    n.MarkNodeAsModified(true);
+                }
+                hwm.Run();
+            }
+        }
+
+        private static void LoadPythonEngine(Assembly assembly)
```

Comment:
Is this deja vu? I vaguely remember seeing this code before. Maybe it's added to DSIronPython2?

Reply:
Should we have a public API to allow customized loading of these engines ?

Reply:
Hi @aparajit-pratap yes deja-vu indeed - see this PR:
https://github.com/DynamoDS/DSIronPython/pull/1

I am unifying both repos so they both load the python engine into separate assembly load contexts.

@pinzart90
1. I'm stopping it from being loaded by putting the binary in the extra folder, the python manager won't load it from there as it's not a package assembly... I guess that distinction means binaries in the bin folder.
2. I think we need that API if you intend to remove access to the `AvailableEngines` property - and not duplicating this code would be great as well.